driver/src/orderbook/filtered_orderbook.rs

```diff
@@ -83,20 +83,20 @@ impl<'a> StableXOrderBookReading for FilteredOrderbookReader<'a> {
                 })
                 .collect(),
         };
-        let user_filtered_orders = token_filtered_orders
-            .into_iter()
-            .filter(|o| {
-                if let Some(user_filter) = self.filter.users.get(&o.account_id) {
-                    match user_filter {
-                        UserOrderFilter::All => false,
-                        UserOrderFilter::OrderIds(ids) => !ids.contains(&o.id),
-                    }
-                } else {
-                    true
+        let user_filtered_orders = token_filtered_orders.into_iter().filter(|o| {
+            if let Some(user_filter) = self.filter.users.get(&o.account_id) {
+                match user_filter {
+                    UserOrderFilter::All => false,
+                    UserOrderFilter::OrderIds(ids) => !ids.contains(&o.id),
                 }
-            })
-            .collect();
-        Ok((state, user_filtered_orders))
+            } else {
+                true
+            }
+        });
+        Ok(util::filter_auction_data(
+            state.into_iter(),
```

Comment:
nit: Doesn't the function takes `IntoIterator`? It think you shouldn't need to call it here if it does.
```suggestion
            state,
```